commit d5337debacc00591b3f81fc3c982b40af7de1ab6
Author: Takashi Iwai <tiwai@suse.de>
Date:   Wed Jan 7 11:41:57 2009 +0100

    ALSA: hda - Add quirk for HP 2230s
    
    Added a quirk for HP 2230s, model=laptop, with AD1984A codec.
    Reference: Novell bnc#461660
    	https://bugzilla.novell.com/show_bug.cgi?id=461660
    
    Signed-off-by: Takashi Iwai <tiwai@suse.de>

commit 11d518e07d700eeb5bcec36bfd5f501e405230dd
Author: Takashi Iwai <tiwai@suse.de>
Date:   Wed Dec 10 10:37:33 2008 +0100

    ALSA: hda - Add quirk for HP6730B laptop
    
    Added model=laptop for HP 6730B laptop with AD1984A codec.
    Reference: Novell bnc#457909
    	https://bugzilla.novell.com/show_bug.cgi?id=457909
    
    Signed-off-by: Takashi Iwai <tiwai@suse.de>
    Cc: stable@kernel.org

commit 65b92e5cbc8acd14ea83190b4d016f765dce6075
Author: Michel Marti <mma@objectxp.com>
Date:   Sat Nov 8 11:33:32 2008 +0100

    ALSA: hda - Add another HP model (6730s) for AD1884A
    
    Added model=laptop for another HP machine (103c:3614) with AD1884A
    codec.
    
    Signed-off-by: Michel Marti <mma@objectxp.com>
    Signed-off-by: Takashi Iwai <tiwai@suse.de>

commit 254248313aed7e6ff295ca21a82ca989b1f69c16
Author: Travis Place <wishie@wishie.net>
Date:   Mon Nov 10 17:56:23 2008 +0100

    ALSA: hda - Make the HP EliteBook 8530p use AD1884A model laptop
    
    Added a QUIRK to patch_analog.c for the HP Elitebook 8530p
    (IDs 0x103c:0x30e7) to use AD1884A model 'laptop' by default.
    Playback and Capture confirmed working.
    
    Signed-off-by: Travis Place <wishie@wishie.net>
    Signed-off-by: Takashi Iwai <tiwai@suse.de>

commit 5695ff44160e62d9193c0201706853bcfe2a077f
Author: Takashi Iwai <tiwai@suse.de>
Date:   Tue Oct 28 15:39:26 2008 +0100

    ALSA: hda - Add another HP model for AD1884A
    
    Added a quirk entry for another HP mobile device with AD1884A codec.
    
    Signed-off-by: Takashi Iwai <tiwai@suse.de>

Adjusted to apply to Debian's 2.6.26 by dann frazier <dannf@hp.com>

diff -urpN linux-source-2.6.26.orig/sound/pci/hda/patch_analog.c linux-source-2.6.26/sound/pci/hda/patch_analog.c
--- linux-source-2.6.26.orig/sound/pci/hda/patch_analog.c	2009-01-29 16:43:51.000000000 -0700
+++ linux-source-2.6.26/sound/pci/hda/patch_analog.c	2009-02-08 16:59:45.000000000 -0700
@@ -3846,6 +3846,11 @@ static const char *ad1884a_models[AD1884
 static struct snd_pci_quirk ad1884a_cfg_tbl[] = {
 	SND_PCI_QUIRK(0x103c, 0x3030, "HP", AD1884A_MOBILE),
 	SND_PCI_QUIRK(0x17aa, 0x20ac, "Thinkpad X300", AD1884A_THINKPAD),
+	SND_PCI_QUIRK(0x103c, 0x3056, "HP", AD1884A_MOBILE),
+       SND_PCI_QUIRK(0x103c, 0x3037, "HP 2230s", AD1884A_LAPTOP),
+	SND_PCI_QUIRK(0x103c, 0x30e6, "HP 6730b", AD1884A_LAPTOP),
+	SND_PCI_QUIRK(0x103c, 0x30e7, "HP EliteBook 8530p", AD1884A_LAPTOP),
+	SND_PCI_QUIRK(0x103c, 0x3614, "HP 6730s", AD1884A_LAPTOP),
 	{}
 };
 
